contracts/mainnet/connectors/morpho-compound/main.sol

```diff
@@ -0,0 +1,440 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+import "./helpers.sol";
+import "./events.sol";
+
+abstract contract MorphoCompound is Helpers, Events {
+	/**
+	 * @dev Deposit ETH/ERC20_Token.
+	 * @notice Deposit a token to Morpho Compound for lending / collaterization.
+	 * @param _tokenAddress The address of underlying token to deposit.(For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param _poolTokenAddress The address of cToken to deposit.(For ETH: cWETH address)
+	 * @param _amount The amount of the token (in underlying) to deposit. (For max: `uint256(-1)`)
+	 * @param _getId ID to retrieve amt.
+	 * @param _setId ID stores the amount of tokens deposited.
+	 */
+	function deposit(
+		address _tokenAddress,
+		address _poolTokenAddress,
+		uint256 _amount,
+		uint256 _getId,
+		uint256 _setId
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		uint256 _amt = getUint(_getId, _amount);
+
+		bool _isETH = _tokenAddress == ethAddr;
+
+		TokenInterface _tokenContract = _isETH
+			? TokenInterface(wethAddr)
+			: TokenInterface(_tokenAddress);
+
+		if (_amt == uint256(-1)) {
+			_amt = _isETH
+				? address(this).balance
+				: _tokenContract.balanceOf(address(this));
+		}
+
+		if (_isETH) convertEthToWeth(_isETH, _tokenContract, _amt);
```

Comment:
Same comment as for Aave

Reply:
Removed the double-check for Aave as well.